kong/runloop/handler.lua

```diff
@@ -925,6 +933,58 @@ return {
         end
       end
 
+      -- clear hop-by-hop request headers:
+      local connection = var.http_connection
+      if connection then
+        local header_names = re_split(connection .. ",", [[\s*,\s*]], "djo")
+        if header_names then
+          for i=1, #header_names do
+            if header_names[i] ~= "" then
+              local header_name = lower(header_names[i])
+              -- some of these are already handled by the proxy module,
+              -- proxy-authorization being an exception that is handled
+              -- below with special semantics.
+              if header_name ~= "close" and
+                 header_name ~= "upgrade" and
+                 header_name ~= "keep-alive" and
+                 header_name ~= "proxy-authorization" then
+                ngx.req.clear_header(header_names[i])
```

Comment:
Maybe we should cache the `clear_header` method as well since it is on a potentially hot code path.

Reply:
Just to note, this was mentioned here as well: https://github.com/Kong/kong/pull/4316#discussion_r257044383